From 302054727f76123c6d4ce996a7410f188ba37c4b Mon Sep 17 00:00:00 2001
Message-Id: <302054727f76123c6d4ce996a7410f188ba37c4b.1380025695.git.minovotn@redhat.com>
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Fri, 30 Aug 2013 09:34:40 +0200
Subject: [PATCH 1/4] Revert "usb-hub: report status changes only once"

RH-Author: Gerd Hoffmann <kraxel@redhat.com>
Message-id: <1377855280-11565-2-git-send-email-kraxel@redhat.com>
Patchwork-id: 53940
O-Subject: [RHEL-6.5 qemu-kvm PATCH 1/1] Revert "usb-hub: report status changes only once"
Bugzilla: 1002888
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

This reverts commit 18143137f8f8924eecea902867546c6beb64ff71.
---
 hw/usb-hub.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/usb-hub.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/hw/usb-hub.c b/hw/usb-hub.c
index 4457024..cf5f401 100644
--- a/hw/usb-hub.c
+++ b/hw/usb-hub.c
@@ -33,7 +33,6 @@ typedef struct USBHubPort {
     USBPort port;
     uint16_t wPortStatus;
     uint16_t wPortChange;
-    uint16_t wPortChange_reported;
 } USBHubPort;
 
 typedef struct USBHubState {
@@ -417,11 +416,8 @@ static int usb_hub_handle_data(USBDevice *dev, USBPacket *p)
             status = 0;
             for(i = 0; i < NUM_PORTS; i++) {
                 port = &s->ports[i];
-                if (port->wPortChange &&
-                    port->wPortChange_reported != port->wPortChange) {
+                if (port->wPortChange)
                     status |= (1 << (i + 1));
-                }
-                port->wPortChange_reported = port->wPortChange;
             }
             if (status != 0) {
                 for(i = 0; i < n; i++) {
-- 
1.7.11.7

